backend/climateconnect_api/tests/test_user_views.py

```diff
@@ -48,19 +77,275 @@ def test_signup_api_success(self):
+        expected_result = {
+            "en": [self.english_user_in_hub_2.id],
+            "de": [self.german_user_in_hub_2.id],
+        }
+        # the function gives back a dict with lang_code as key and a query as value
+        # the query needs to be evaluated first
+        for key_result, result_query in result.items():
+            result[key_result] = list(result_query)
+        self.assertCountEqual(result.items(), expected_result.items())
+
+        # for international
+        is_in_hub = False
+        # location_id is 0 for international
+        loc = 0
+        result = fetch_user_info_for_weekly_recommendations(loc, is_in_hub)
+        expected_result = {
+            "en": [self.english_user_no_hub.id, self.no_language_user_no_hub.id],
+            "de": [self.german_user_no_hub.id],
+        }
+        # the function gives back a dict with lang_code as key and a query as value
+        # the query needs to be evaluated first
+        for key_result, result_query in result.items():
+            result[key_result] = list(result_query)
+        self.assertCountEqual(result, expected_result)
+
+    def test_send_email(self):
+        """This function mimics the function schedule_weekly_recommendations_email() in climateconnect_api.tasks.py and sends emails to the mailjet_admin_email adress set in .backend_env if variable sandbox_mode is set"""
+        # it sends emails for each hub/international and for each language, so a maximum number of 6 emails
+        # to turn email sending on, set sandbox_mode to False
+        sandbox_mode = False
+        # to determine how many emails get sent, please set max_emails_sent variable to your desired number
+        max_emails_sent = 2
+
+        max_entities = 3
+        timespan_start = timezone.now() - timedelta(days=7)
+
+        all_locations_in_hubs = list(
+            Location.objects.filter(hub_location__hub_type=Hub.LOCATION_HUB_TYPE)
+            .values_list("id", flat=True)
+            .distinct()
+        )
+        # "0" acts as a flag for the international recommendations email
+        all_locations_in_hubs.append(0)
+        for location_id in all_locations_in_hubs:
+
+            is_in_hub = location_id != 0
+
+            entity_ids = fetch_entities_for_weekly_recommendations(
+                max_entities, timespan_start, location_id, is_in_hub
+            )
+
+            if entity_ids:
+                lang_codes = list(
+                    Language.objects.values_list("language_code", flat=True).distinct()
+                )
+                for lang_code in lang_codes:
+                    mailjet_global_vars = (
+                        create_global_variables_for_weekly_recommendations(
+                            entity_ids, lang_code, is_in_hub
+                        )
+                    )
+                    if max_emails_sent < 1:
+                        break
+                    result = process_user_info_and_send_weekly_recommendations(
```

Comment:
You can use mock here so we don't call third party mailjet services here. https://docs.python.org/3/library/unittest.mock.html

I can help with mocking data if you like

Reply:
We should never write tests for third party services. We can always mock those methods and test code that we wrote.

Reply:
Thanks, i separated this now from the unittests.